src/devices/Bmxx80/Bme680.cs

```diff
@@ -0,0 +1,235 @@
+        public bool HasNewData => ReadHasNewData();
+
+        /// <summary>
+        /// Gets the humidity in %rH (percentage relative humidity).
+        /// </summary>
+        public double Humidity => ReadHumidity();
+
+        /// <summary>
+        /// Get the pressure in Pa (Pascal).
+        /// </summary>
+        public double Pressure => ReadPressure();
+
+        /// <summary>
+        /// Gets the <see cref="Temperature"/>.
+        /// </summary>
+        public Temperature Temperature => ReadTemperature();
+
+        /// <summary>
+        /// Default I2C bus address.
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x76;
+
+        /// <summary>
+        /// Secondary I2C bus address.
+        /// </summary>
+        public const byte SecondaryI2cAddress = 0x77;
+
+        /// <summary>
+        /// The expected chip ID of the BME68x product family.
+        /// </summary>
+        private const byte DeviceId = 0x61;
+
+        /// <summary>
+        /// Initialize a new instance of the <see cref="Bme680"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Bme680(I2cDevice i2cDevice)
+            : base(DeviceId, i2cDevice)
+        {
+            _calibrationData = new Bme680CalibrationData();
+            _calibrationData.ReadFromDevice(this);
+            _communicationProtocol = CommunicationProtocol.I2c;
+            _controlRegister = (byte)Bme680Register.CONTROL;
+        }
+
+        /// <summary>
+        /// Set the humidity sampling.
+        /// </summary>
+        /// <param name="sampling">The <see cref="Sampling"/> to set.</param>
+        public void SetHumiditySampling(Sampling sampling)
+        {
+            var register = (byte)Bme680Register.CONTROL_HUM;
+            byte read = Read8BitsFromRegister(register);
+
+            // Clear first 3 bits.
+            var cleared = (byte)(read & 0b_1111_1000);
+
+            _i2cDevice.Write(new[] { register, (byte)(cleared | (byte)sampling) });
+        }
+
+        /// <summary>
+        /// Sets the power mode to the given mode
+        /// </summary>
+        /// <param name="powerMode">The <see cref="Bme680PowerMode"/> to set.</param>
+        public void SetPowerMode(Bme680PowerMode powerMode)
+        {
+            byte read = Read8BitsFromRegister(_controlRegister);
+
+            // Clear first 2 bits.
+            var cleared = (byte)(read & 0b_1111_1100);
+
+            _i2cDevice.Write(new[] { _controlRegister, (byte)(cleared | (byte)powerMode) });
+        }
+
+        /// <summary>
+        /// Compensates the humidity.
+        /// </summary>
+        /// <param name="temperature">The temperature to use.</param>
+        /// <param name="adcHumidity">The humidity value read from the device.</param>
+        /// <returns>The percentage relative humidity.</returns>
+        private double CompensateHumidity(double temperature, int adcHumidity)
```

Comment:
I suppose these compensation methods are very different to the ones in the 280Base?

Reply:
I believe they are. The only common one is temperature.